wolfssl/wolfcrypt/types.h

```diff
@@ -100,7 +100,8 @@
     /* These platforms have 64-bit CPU registers.  */
     #if (defined(__alpha__) || defined(__ia64__) || defined(_ARCH_PPC64) || \
          defined(__mips64)  || defined(__x86_64__) || defined(_M_X64)) || \
-         defined(__aarch64__) || defined(__sparc64__)
+         defined(__aarch64__) || defined(__sparc64__) || \
+        (defined(__SIZEOF_POINTER__) && (__SIZEOF_POINTER__ == 8))
```

Comment:
Is it possible sizeof_pointer can be 8 on systems with 32bit general purpose registers?  That is, is there a more specific define that cover HiFive here?

Reply:
@dgarske @toddouska perhaps we can use `__riscv_xlen` for this purpose.

```
riscv64-unknown-linux-gnu-gcc -dM -E -x c /dev/null |grep riscv_xlen 
#define __riscv_xlen 64

riscv32-unknown-elf-gcc -dM -E -x c /dev/null |grep riscv_xlen
#define __riscv_xlen 32
```

Reply:
Tested the following, compiles well (both with native gcc and cross-compiled as per instructions).


```diff
--- a/wolfssl/wolfcrypt/types.h
+++ b/wolfssl/wolfcrypt/types.h
@@ -101,7 +101,7 @@
     #if (defined(__alpha__) || defined(__ia64__) || defined(_ARCH_PPC64) || \
          defined(__mips64)  || defined(__x86_64__) || defined(_M_X64)) || \
          defined(__aarch64__) || defined(__sparc64__) || \
-        (defined(__SIZEOF_POINTER__) && (__SIZEOF_POINTER__ == 8))
+        (defined(__riscv_xlen) && (__riscv_xlen == 64))
         typedef word64 wolfssl_word;
         #define WC_64BIT_CPU
     #elif (defined(sun) || defined(__sun)) && \
```